OOP Library: 2 - Decorate a class

lmunozm1702 opened the pull request:
## In this PR, I do:

### Interface: create a class Nameable.
### Turn the Person class to Nameable.
### Prepare base Decorator
### Prepare CapitalizeDecorator and TrimmerDecorator

lmunozm1702 committed:
Create Decorator for Nameable

- Create Nameable class and his decorators
- Update Person class to set inheritance from Nameable

lmunozm1702 committed:
Fix linter errors

mckent05 commented on nameable.rb:
```diff
@@ -0,0 +1,32 @@
+  end
+end
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    return @nameable.correct_name[0..9] if @nameable.correct_name.length > 10
+
+    @nameable.correct_name
+  end
+end
```
- According to the project requirements, each of the individual classes should be in a different file not in the same file as the `Nameable` class. Please fix this 🙏

lmunozm1702 replied:
Oh, OK, I was missed because, in the lesson documentation, the Decorators and the "main" class are in the same file, I will change it right now! thanks for the advice.

mckent05 reviewed (changes requested):
Hi @lmunozm1702, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
![almost there](https://user-images.githubusercontent.com/73607512/207265056-d056ea78-77c5-4b66-802e-694a5ed4ecf3.gif)

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

lmunozm1702 committed:
Fix code review issue

According to the project requirements, each of the individual classes should be in a different file not in the same file as the Nameable class.

lmunozm1702 committed:
Update persons.rb

rotimiazeez reviewed (approved):
Hi @lmunozm1702 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good job with the project implementation 👍🏽 
- No linter Error 👍🏽

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
